source/include/dqm4hep/AnalysisHelper.h

```diff
@@ -0,0 +1,65 @@
+/// \file AnalysislHelper.h
+/*
+ *
+ * AnalysisHelper.h header template automatically generated by a class generator
+ * Creation date : ???
+ *
+ * This file is part of DQM4HEP libraries.
+ *
+ * DQM4HEP is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * based upon these libraries are permitted. Any copy of these libraries
+ * must include this copyright notice.
+ *
+ * DQM4HEP is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with DQM4HEP.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Tom Coates
+ * @copyright CNRS , IPNL
+ */
+
+#ifndef DQM4HEP_ANALYSISHELPER_H
+#define DQM4HEP_ANALYSISHELPER_H
+
+// Need to include ROOT headers so that it can interpret things like TH1 etc.
+
+// -- dqm4hep headers
+#include <dqm4hep/StatusCodes.h>
+#include <dqm4hep/Internal.h>
+#include <dqm4hep/Logging.h>
+#include <dqm4hep/MonitorElement.h>
+
+// -- root headers
+#include <TH1.h>
+#include <TMath.h>
+
+namespace dqm4hep {
+
+  namespace core {
+
+    /** analysis helper class
+     */
+    class AnalysisHelper
```

Comment:
Here you should have separate functions for mean, mean90, rms90, etc ...
Simply call them `AnalysisHelper::mean` or `AnalysisHelper::mean90` and return the value

Reply:
The reason I didn't define separate functions was to allow more flexibility, so you could make tests like mean75 or mean80, as well as mean90. If that's not desired, then I can revert to separate functions, which will allow me to remove the m_percentage variable and simplify some of the logic.

Reply:
Having a function like e.g `mean(element, 75)` is a nice thing to have. Could you just for convenience add one function mean90/rms90 as they are standard in HEP now simply call `mean(element, 90)` in the implementation.

About the string that is passed in the function mainHelper, I think the string parsing/evaluation should belong to the qtest itself and shouldn't appear here